pkg/apis/crd/v1alpha2/types.go

```diff
@@ -543,3 +543,88 @@ type TrafficControlList struct {
 
 	Items []TrafficControl `json:"items"`
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// BGPPolicy defines BGP configuration applied to Nodes.
+type BGPPolicy struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard metadata of the object.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec BGPPolicySpec `json:"spec"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type BGPPolicyList struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []BGPPolicy `json:"items"`
+}
+
+// BGPPolicySpec defines the specification for a BGP policy.
+type BGPPolicySpec struct {
+	// NodeSelector is utilized to select Nodes to which the policy is applied.
+	NodeSelector *metav1.LabelSelector `json:"nodeSelector"`
+
+	// LocalASN is the AS number used by BGP protocol.
+	LocalASN int32 `json:"localASN"`
+
+	// ListenPort is the port on which BGP protocol listen.
+	ListenPort *int32 `json:"listenPort,omitempty"`
+
+	// Advertisements selects IPs to be advertised to external BGP peers.
+	Advertisements Advertisements `json:"advertisements"`
+
+	// BGPPeers are the list of external BGP peers.
+	BGPPeers []BGPPeer `json:"bgpPeers"`
+}
+
+type Advertisements struct {
+	ServiceIPs *ServiceAdvertisement `json:"serviceIPs,omitempty"`
+
+	PodIPs *PodAdvertisement `json:"podIPs,omitempty"`
+
+	// Determine whether to advertise local Egress IPs.
+	EgressIPs bool `json:"egressIPs,omitempty"`
+}
+
+type ServiceAdvertisement struct {
+	// Determine whether to advertise ClusterIPs.
+	ClusterIPs bool `json:"clusterIPs,omitempty"`
+
+	// Determine whether to advertise ExternalIPs.
+	ExternalIPs bool `json:"externalIPs,omitempty"`
+
+	// Determine whether to advertise LoadBalancerIPs.
+	LoadBalancerIPs bool `json:"loadBalancerIPs,omitempty"`
+}
+
+type PodAdvertisement struct {
+	// Determine whether to advertise local Pod CIDR.
+	Enable bool `json:"enable,omitempty"`
+}
+
+type BGPPeer struct {
+	// Address is the IP address of the external BGP peer.
+	Address string `json:"address"`
+
+	// Port is the port number of the external BGP peer.
+	Port *int32 `json:"port,omitempty"`
+
+	// ASN is the ASN of the external BGP peer.
+	ASN int32 `json:"asn"`
+
+	// AuthSecretRef is a reference to a Secret storing the authentication password of the external BGP peer.
+	AuthSecretRef *string `json:"authSecretRef,omitempty"`
+
+	// GracefulRestart defines graceful restart parameters which are negotiated with the external BGP peer. If this is
+	// set, the graceful restart is disabled.
+	GracefulRestart *int32 `json:"gracefulRestart,omitempty"`
```

Comment:
This field is a bit confusing because you mention "parameters" in the comment, but this is an integer value. What does the value represent?

Reply:
I have refined this comment. The original one was not good.